[621] Improve Software Management Architecture doc (#722)
thin-edge/thin-edge.io: docs/src/architecture/software-management.md

## Software management components

Software Management uses the following 3 components to perform software operations:
**Cloud Mapper**, **Agent**, and **Software Management Plugin**.

> ![Software management components](./images/software-management.png)

Author: ```suggestion
You can find the diagrams which explain how those 3 components interact with each other from [Software Management Agent Specification](https://thin-edge.github.io/thin-edge.io-specs/software-management/sm-agent.html).
```

## Software management components

> Software Management uses following the 3 components to perform software operations:

Reviewer: ```suggestion
Software Management uses the following 3 components to perform software operations:
```

**Cloud Mapper**, **Agent**, and **Software Management Plugin**.

![Software management components](./images/software-management.png)

### Cloud Mapper

The **Cloud Mapper** converts from/to cloud-specific format to/from cloud-agnostic format.
It communicates with the dedicated IoT cloud platform and the **Tedge Agent**.

### Tedge Agent

> The **Tedge Agent** addresses cloud-agnostic software management operations along with the **Cloud Mapper**,
> e.g. listing current installed software list, software update, software removal.

Reviewer: Shouldn't the Cloudmapper be removed from this sentence?
```suggestion
The **Tedge Agent** addresses cloud-agnostic software management operations e.g. listing current installed software list, software update, software removal.
```

> Also, the Tedge Agent calls the **SM Plugin(s)** to execute an action defined by a received operation.

Reviewer: ```suggestion
Also, the Tedge Agent calls an **SM Plugin(s)** to execute an action defined by a received operation.
```

> The key point is that the Tedge Agent is always generic in cloud platforms and software types.

Reviewer: How about this?
```suggestion
The key point is that the **Tedge Agent** is always generic in cloud platforms and software types and **Tedge Mapper** handles cloud specific actions.
```

Author: A bit correction:
```suggestion
The key points are that the **Tedge Agent** is always generic in cloud platforms and software types, and **Cloud Mapper** handles cloud-specific actions.
```
